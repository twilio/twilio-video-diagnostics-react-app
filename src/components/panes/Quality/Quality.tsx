import { ExcellentQuality } from './ExcellentQuality/ExcellentQuality';
import { PoorQuality } from './PoorQuality/PoorQuality';
import { getQualityScore } from './getQualityScore/getQualityScore';
import { useAppStateContext } from '../../AppStateProvider/AppStateProvider';
import { QualityModal } from './QualityModal/QualityModal';
import { useState } from 'react';

export enum QualityScore {
  Poor,
  Suboptimal,
  Good,
  Excellent,
}

export function Quality() {
  const { state } = useAppStateContext();
  const [isModalOpen, setIsModalOpen] = useState(false);
  const { totalQualityScore, latency, jitter, packetLoss, bitrate } = getQualityScore(
    state.preflightTest.report,
    state.bitrateTest.report
  );

  return (
    <>
      <QualityModal
        isModalOpen={isModalOpen}
        setIsModalOpen={setIsModalOpen}
        jitter={jitter}
        latency={latency}
        packetLoss={packetLoss}
        bitrate={bitrate}
      />
      {totalQualityScore === QualityScore.Good || totalQualityScore === QualityScore.Excellent ? (
        <ExcellentQuality quality={totalQualityScore} openModal={() => setIsModalOpen(true)} />
      ) : (
        <PoorQuality quality={totalQualityScore} openModal={() => setIsModalOpen(true)} />
      )}
    </>
  );
}
