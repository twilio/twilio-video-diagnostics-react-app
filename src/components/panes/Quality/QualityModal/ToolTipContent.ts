export const toolTipContent = {
  roundTripTime: `Round-trip time (RTT) is the time a packet of data takes to go from sender to receiver and back again. High RTT can cause the video to lag, freeze, or slow down. This is usually attributed to slow or overloaded networks.`,
  jitter: `Jitter is the fluctuation of latency over time. Due to network congestion, data can take longer to travel. High jitter typically results in short bursts of video freezing, slow video, and choppy audio.`,
  packetLoss: `Packet loss is caused when data fails to reach its destination. High packet loss results in video freezing, slow video, and choppy audio. Packet loss is usually caused by overloaded routers or high CPU load on the machine.`,
  bitrate: `Bitrate is the way we measure the speed of the upload and download transfers. If a sender uploads data too quickly and exceeds the transfer speed, the network gets overwhelmed and can cause issues for everyone. Symptoms of this can be frozen video, downscaled video, frame-rate drops, choppy audio, and potentially dropped connections.`,
};
