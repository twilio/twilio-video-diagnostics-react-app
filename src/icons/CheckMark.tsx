export function CheckMark() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M10 18C14.4183 18 18 14.4183 18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18ZM9.7103 12.7047L13.7062 8.71128C13.8943 8.52328 14 8.26831 14 8.00244C14 7.73657 13.8943 7.4816 13.7062 7.29361C13.5181 7.10561 13.263 7 12.9969 7C12.7309 7 12.4758 7.10561 12.2877 7.29361L9.00103 10.5882L7.71234 9.29032C7.52423 9.10232 7.2691 8.99672 7.00307 8.99672C6.73703 8.99672 6.4819 9.10232 6.2938 9.29032C6.10568 9.47831 6 9.73328 6 9.99915C6 10.265 6.10568 10.52 6.2938 10.708L8.29175 12.7047C8.38462 12.7983 8.4951 12.8725 8.61684 12.9232C8.73857 12.9739 8.86914 13 9.00103 13C9.1329 13 9.26347 12.9739 9.3852 12.9232C9.50694 12.8725 9.61742 12.7983 9.7103 12.7047Z"
        fill="#0E7C3A"
      />
    </svg>
  );
}
