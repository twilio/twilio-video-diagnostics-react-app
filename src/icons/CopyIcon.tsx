export function CopyIcon() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M14.6812 3.59414C14.6188 2.97958 14.0998 2.5 13.4688 2.5H3.71875L3.59414 2.50629C2.97958 2.5687 2.5 3.08772 2.5 3.71875V13.4688L2.50629 13.5934C2.5687 14.2079 3.08772 14.6875 3.71875 14.6875H5.3125V16.0938C5.3125 16.8704 5.9421 17.5 6.71875 17.5H16.0938C16.8704 17.5 17.5 16.8704 17.5 16.0938V6.71875C17.5 5.9421 16.8704 5.3125 16.0938 5.3125H14.6875V3.71875L14.6812 3.59414ZM13.75 5.3125V3.71875C13.75 3.58561 13.6575 3.47408 13.5332 3.44493L13.4688 3.4375H3.71875C3.58561 3.4375 3.47408 3.53001 3.44493 3.65426L3.4375 3.71875V13.4688C3.4375 13.6019 3.53001 13.7134 3.65426 13.7426L3.71875 13.75H5.3125V6.71875C5.3125 5.9421 5.9421 5.3125 6.71875 5.3125H13.75ZM6.71875 6.25H16.0938C16.3526 6.25 16.5625 6.45987 16.5625 6.71875V16.0938C16.5625 16.3526 16.3526 16.5625 16.0938 16.5625H6.71875C6.45987 16.5625 6.25 16.3526 6.25 16.0938V6.71875C6.25 6.45987 6.45987 6.25 6.71875 6.25Z"
        fill="#606B85"
      />
    </svg>
  );
}
