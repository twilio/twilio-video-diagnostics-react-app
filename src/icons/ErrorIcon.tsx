export function ErrorIcon() {
  return (
    <div style={{ display: 'inline-block', marginRight: '10px' }}>
      <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M13.7461 1.52332L22.4767 10.2539C23.4411 11.2183 23.4411 12.7818 22.4767 13.7462L13.7461 22.4768C12.7818 23.4411 11.2182 23.4411 10.2539 22.4768L1.52327 13.7462C0.55891 12.7818 0.55891 11.2183 1.52327 10.2539L10.2539 1.52332C11.2182 0.55896 12.7818 0.55896 13.7461 1.52332ZM13.4 16.2C13.4 16.9732 12.7732 17.6 12 17.6C11.2268 17.6 10.6 16.9732 10.6 16.2C10.6 15.4269 11.2268 14.8 12 14.8C12.7732 14.8 13.4 15.4269 13.4 16.2ZM11.9856 7.80005H12.0142C12.8694 7.81042 13.5156 8.62935 13.3826 9.54159L12.7839 12.6744C12.7269 13.0891 12.3943 13.4 12.0047 13.4H11.9951C11.6055 13.4 11.2729 13.0891 11.2159 12.6744L10.6172 9.54159C10.4842 8.62935 11.1399 7.80005 11.9856 7.80005Z"
          fill="#D61F1F"
        />
      </svg>
    </div>
  );
}
