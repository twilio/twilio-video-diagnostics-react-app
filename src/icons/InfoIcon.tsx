export function InfoIcon() {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M18 10C18 5.58172 14.4183 2 10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C14.4183 18 18 14.4183 18 10ZM3.25 10C3.25 6.27208 6.27208 3.25 10 3.25C13.7279 3.25 16.75 6.27208 16.75 10C16.75 13.7279 13.7279 16.75 10 16.75C6.27208 16.75 3.25 13.7279 3.25 10ZM9 10C9 9.44772 9.44772 9 10 9C10.5523 9 11 9.44772 11 10V13C11 13.5523 10.5523 14 10 14C9.44772 14 9 13.5523 9 13V10ZM10 8C10.5523 8 11 7.55228 11 7C11 6.44772 10.5523 6 10 6C9.44772 6 9 6.44772 9 7C9 7.55228 9.44772 8 10 8Z"
        fill="#606B85"
      />
    </svg>
  );
}
