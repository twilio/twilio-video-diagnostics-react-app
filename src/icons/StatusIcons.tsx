export const SuccessStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect width="20" height="20" fill="white" fillOpacity="0.01" />
    <circle cx="10" cy="10" r="6" fill="#14B053" />
  </svg>
);

export const ErrorStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0H20V20H0V0Z" fill="white" fillOpacity="0.01" />
    <rect x="10.0713" y="3" width="10" height="10" transform="rotate(45 10.0713 3)" fill="#D61F1F" />
  </svg>
);

export const WarningStatus = () => (
  <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0H20V20H0V0Z" fill="white" fillOpacity="0.01" />
    <rect x="10.0713" y="3" width="10" height="10" transform="rotate(45 10.0713 3)" fill="#F5A623" />
  </svg>
);
