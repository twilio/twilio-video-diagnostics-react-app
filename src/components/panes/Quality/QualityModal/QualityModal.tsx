import {
  createStyles,
  makeStyles,
  Typography,
  Dialog,
  DialogTitle,
  DialogContent,
  IconButton,
  Table,
  TableHead,
  TableBody,
  TableRow,
  TableCell,
  Tooltip,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import { ErrorStatus, SuccessStatus } from '../../../../icons/StatusIcons';
import { QualityScore } from '../Quality';
import { toolTipContent } from './ToolTipContent';
import { InfoIcon } from '../../../../icons/InfoIcon';

const useStyles = makeStyles((theme) =>
  createStyles({
    modal: {
      width: '100%',
    },
    content: {
      padding: '3em',
    },
    closeButton: {
      position: 'absolute',
      right: theme.spacing(1),
      top: theme.spacing(1),
      color: theme.palette.grey[500],
    },
    iconContainer: {
      display: 'flex',
      alignItems: 'center',
      '& div': {
        display: 'flex',
        alignItems: 'center',
      },
      '& svg': {
        marginRight: '0.3em',
      },
    },
  })
);

interface QualityModalProps {
  isModalOpen: boolean;
  setIsModalOpen: (isModalOpen: boolean) => void;
  jitter: { average?: string; max?: string; qualityScore: QualityScore };
  latency: { average?: string; max?: string; qualityScore: QualityScore };
  packetLoss: { average?: string; max?: string; qualityScore: QualityScore };
  bitrate: { average?: string; max?: string; min?: string; qualityScore: QualityScore };
}

export function QualityModal({ isModalOpen, setIsModalOpen, latency, jitter, packetLoss, bitrate }: QualityModalProps) {
  const classes = useStyles();

  return (
    <>
      <Dialog open={isModalOpen} onClose={() => setIsModalOpen(false)} classes={{ paper: classes.modal }}>
        <DialogTitle disableTypography>
          <Typography variant="h3">Detailed quality information</Typography>
          <IconButton aria-label="close" className={classes.closeButton} onClick={() => setIsModalOpen(false)}>
            <CloseIcon />
          </IconButton>
        </DialogTitle>
        <DialogContent dividers className={classes.content}>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>
                  <Typography variant="body1">
                    <strong>Network Info</strong>
                  </Typography>
                </TableCell>
                <TableCell>
                  <Typography variant="body1">
                    <strong>Score</strong>
                  </Typography>
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    <Tooltip title={toolTipContent.latency} interactive leaveDelay={250}>
                      <div>
                        <InfoIcon />
                      </div>
                    </Tooltip>
                    <Typography variant="body1">
                      <strong>Latency (ms) avg/max</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {latency.qualityScore === QualityScore.Good || latency.qualityScore === QualityScore.Excellent ? (
                      <SuccessStatus />
                    ) : (
                      <ErrorStatus />
                    )}
                    <Typography variant="body1">{`${latency.average} / ${latency.max}`}</Typography>
                  </div>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    <Tooltip title={toolTipContent.jitter} interactive leaveDelay={250}>
                      <div>
                        <InfoIcon />
                      </div>
                    </Tooltip>
                    <Typography variant="body1">
                      <strong>Jitter (s) avg/max</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {jitter.qualityScore === QualityScore.Good || jitter.qualityScore === QualityScore.Excellent ? (
                      <SuccessStatus />
                    ) : (
                      <ErrorStatus />
                    )}
                    <Typography variant="body1">{`${jitter.average} / ${jitter.max}`}</Typography>
                  </div>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    <Tooltip title={toolTipContent.packetLoss} interactive leaveDelay={250}>
                      <div>
                        <InfoIcon />
                      </div>
                    </Tooltip>
                    <Typography variant="body1">
                      <strong>Packet loss avg/max</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {packetLoss.qualityScore === QualityScore.Good ||
                    packetLoss.qualityScore === QualityScore.Excellent ? (
                      <SuccessStatus />
                    ) : (
                      <ErrorStatus />
                    )}
                    <Typography variant="body1">{`${packetLoss.average}% / ${packetLoss.max}%`}</Typography>
                  </div>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>
                  <div className={classes.iconContainer}>
                    <Tooltip title={toolTipContent.bitrate} interactive leaveDelay={250}>
                      <div>
                        <InfoIcon />
                      </div>
                    </Tooltip>
                    <Typography variant="body1">
                      <strong>Bitrate (kbps) max/avg/min</strong>
                    </Typography>
                  </div>
                </TableCell>
                <TableCell>
                  <div className={classes.iconContainer}>
                    {bitrate.qualityScore === QualityScore.Good || bitrate.qualityScore === QualityScore.Excellent ? (
                      <SuccessStatus />
                    ) : (
                      <ErrorStatus />
                    )}
                    <Typography variant="body1">{`${bitrate.max} / ${bitrate.average} / ${bitrate.min}`}</Typography>
                  </div>
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </DialogContent>
      </Dialog>
    </>
  );
}
