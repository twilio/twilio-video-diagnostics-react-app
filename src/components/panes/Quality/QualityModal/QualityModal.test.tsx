import { shallow } from 'enzyme';
import { QualityScore } from '../Quality';
import { QualityModal } from './QualityModal';
import { ErrorStatus, SuccessStatus } from '../../../../icons/StatusIcons';

const mockJitter = { average: '0', max: '3', qualityScore: QualityScore.Excellent };
const mockLatency = { average: '85.2', max: '192', qualityScore: QualityScore.Excellent };
const mockPacketLoss = { average: '0', max: '2', qualityScore: QualityScore.Excellent };
const mockBitrate = { average: '11,608.06', max: '8,333.98', min: '3,612.67', qualityScore: QualityScore.Excellent };

describe('the QualityModal component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <QualityModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        jitter={mockJitter}
        latency={mockLatency}
        packetLoss={mockPacketLoss}
        bitrate={mockBitrate}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should use the success icons for "good" and "excellent" quality scores', () => {
    const wrapper = shallow(
      <QualityModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        jitter={mockJitter}
        latency={mockLatency}
        packetLoss={mockPacketLoss}
        bitrate={mockBitrate}
      />
    );

    expect(wrapper.find(SuccessStatus).exists()).toBe(true);
    expect(wrapper.find(ErrorStatus).exists()).toBe(false);
  });

  it('should use the error icons for "poor" and "suboptimal" quality scores', () => {
    const wrapper = shallow(
      <QualityModal
        isModalOpen={true}
        setIsModalOpen={jest.fn()}
        jitter={mockJitter}
        latency={{ ...mockLatency, qualityScore: QualityScore.Poor }}
        packetLoss={mockPacketLoss}
        bitrate={{ ...mockBitrate, qualityScore: QualityScore.Suboptimal }}
      />
    );
    expect(wrapper.find(ErrorStatus).exists()).toBe(true);
  });
});
