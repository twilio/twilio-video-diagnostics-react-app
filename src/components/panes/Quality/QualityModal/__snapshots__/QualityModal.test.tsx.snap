// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the QualityModal component should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Dialog))
    classes={
      Object {
        "paper": "makeStyles-modal-1",
      }
    }
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(DialogTitle))
      disableTypography={true}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h3"
      >
        Detailed quality information
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="close"
        className="makeStyles-closeButton-3"
        onClick={[Function]}
      >
        <Memo(ForwardRef(CloseIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))
      className="makeStyles-content-2"
      dividers={true}
    >
      <WithStyles(ForwardRef(Table))>
        <WithStyles(ForwardRef(TableHead))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                <strong>
                  Network Info
                </strong>
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(Typography))
                variant="body1"
              >
                <strong>
                  Score
                </strong>
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableBody))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <WithStyles(ForwardRef(Tooltip))
                  enterTouchDelay={0}
                  interactive={true}
                  leaveDelay={250}
                  leaveTouchDelay={15000}
                  title="Round-trip time (RTT) is the time a packet of data takes to go from sender to receiver and back again. High RTT can cause the video to lag, freeze, or slow down. This is usually attributed to slow or overloaded networks."
                >
                  <div>
                    <InfoIcon />
                  </div>
                </WithStyles(ForwardRef(Tooltip))>
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    RTT (ms) avg/max
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  85.2 / 192 (Excellent)
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <WithStyles(ForwardRef(Tooltip))
                  enterTouchDelay={0}
                  interactive={true}
                  leaveDelay={250}
                  leaveTouchDelay={15000}
                  title="Jitter is the fluctuation of latency over time. Due to network congestion, data can take longer to travel. High jitter typically results in short bursts of video freezing, slow video, and choppy audio."
                >
                  <div>
                    <InfoIcon />
                  </div>
                </WithStyles(ForwardRef(Tooltip))>
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    Jitter (s) avg/max
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  0 / 3 (Excellent)
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <WithStyles(ForwardRef(Tooltip))
                  enterTouchDelay={0}
                  interactive={true}
                  leaveDelay={250}
                  leaveTouchDelay={15000}
                  title="Packet loss is caused when data fails to reach its destination. High packet loss results in video freezing, slow video, and choppy audio. Packet loss is usually caused by overloaded routers or high CPU load on the machine."
                >
                  <div>
                    <InfoIcon />
                  </div>
                </WithStyles(ForwardRef(Tooltip))>
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    Packet loss avg/max
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  0% / 2% (Excellent)
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <WithStyles(ForwardRef(Tooltip))
                  enterTouchDelay={0}
                  interactive={true}
                  leaveDelay={250}
                  leaveTouchDelay={15000}
                  title="Bitrate is the way we measure the speed of the upload and download transfers. If a sender uploads data too quickly and exceeds the transfer speed, the network gets overwhelmed and can cause issues for everyone. Symptoms of this can be frozen video, downscaled video, frame-rate drops, choppy audio, and potentially dropped connections."
                >
                  <div>
                    <InfoIcon />
                  </div>
                </WithStyles(ForwardRef(Tooltip))>
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  <strong>
                    Bitrate (kbps) avg/max
                  </strong>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
            <WithStyles(ForwardRef(TableCell))>
              <div
                className="makeStyles-iconContainer-4"
              >
                <SuccessStatus />
                <WithStyles(ForwardRef(Typography))
                  variant="body1"
                >
                  8,333.98 / 11,608.06 (Excellent)
                </WithStyles(ForwardRef(Typography))>
              </div>
            </WithStyles(ForwardRef(TableCell))>
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </WithStyles(ForwardRef(DialogContent))>
  </WithStyles(ForwardRef(Dialog))>
</Fragment>
`;
